### PR TITLE
Document Snaps notifications

### LINKED ISSUE #1261: Add Snaps notifications feature page
Add topic under Snaps > Features describing how to display notifications using `snap_notify`. Link to the [notifications example package](https://github.com/MetaMask/snaps/tree/main/packages/examples/packages/notifications). The notifications card on the Snaps intro page should link to this feature page.

### PR DESCRIPTION
**Pull request by alexandratran**
# Description

Add feature page for Snaps notifications.

## Issue(s) fixed

Fixes #1261

## Preview

https://docs.metamask.io/1261-notifications/snaps/features/notifications/

## Checklist

Complete this checklist before merging your PR:

- [x] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page.
- [ ] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

File: snaps/index.mdx [hunk 1]
```diff
@@ -32,9 +32,9 @@ The following Snaps features are available in the stable version of MetaMask:
     },
     {
       icon: require("./assets/features/notifications.png").default,
-      href: "reference/snaps-api#snap_notify",
+      href: "features/notifications",
       title: "Notifications",
-      description: "Notify users directly in MetaMask, or in their OS."
+      description: "Notify users directly in MetaMask, or natively in their browser."
```
Comment (Montoya) [inline snaps/index.mdx:37]: I do not know if it is correct to change the last word. Browser notification APIs are displayed in the OS. They are not displayed in the browser. The Chrome spec says they are displayed in the system tray: https://developer.chrome.com/docs/extensions/reference/api/notifications

File: snaps/reference/snaps-api.md [hunk 2]
```diff
@@ -833,25 +833,16 @@ await snap.request({
 
 ## `snap_notify`
 
-Displays a notification in MetaMask or natively in the browser.
+Displays a [notification](../features/notifications.md) in MetaMask or natively in the browser.
```
Comment (Montoya) [inline snaps/reference/snaps-api.md:836]: Last word is probably a mistake, should be OS.

**Review by Montoya — CHANGES_REQUESTED** [review]

File: snaps/features/notifications.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+### 2. Create a notification
+
+Create a notification by calling [`snap_notify`](../reference/snaps-api.md#snap_notify), which takes
+a notification `type` and `message`.
+Specify `type: "inApp"` to display the notification in the MetaMask UI, and `type: "native"` to
+display the notification in the user's browser.
+
+:::note
+We recommend using `type: "inApp"` because there's no guarantee that native notifications are
```
[the remark this answers is not in the document]
Comment (alexandratran) [reply snaps/features/notifications.md:32]: Can you expand on this?

**Review by Montoya — APPROVED** [review]

**Review by bgravenorst — APPROVED** [review]
lgtm

**Pull request merged**